ext/funzip/unzipper.cpp

```
@@ -0,0 +1,78 @@
+#include <zip.h>
+#include <stdio.h>
+#include <string>
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+
+#include "unzipper.h"
+#include "util.h"
+
+using namespace std;
+
+Unzipper::Unzipper(string archivePath) {
+  archive = TRY(zip_open(archivePath.c_str(), ZIP_CHECKCONS, nullptr), "zip_open");
+}
+
+Unzipper::~Unzipper() {
+  TRY0(zip_close(archive), "zip_close");
```

@rdwampler good call. In this case, if releasing the resource fails, there's not much we can do, and nothing is relying on this to succeed either, so we might as well silently fail. I've removed the `TRY0`'s from the destructors.